src/sinks/datadog/logs.rs

```diff
@@ -0,0 +1,85 @@
+use crate::{
+    sinks::util::{
+        self,
+        tcp::{tcp_healthcheck, TcpSink},
+        tls::{TlsOptions, TlsSettings},
+        uri::UriSerde,
+        Encoding,
+    },
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use bytes::Bytes;
+use futures::{stream::iter_ok, Sink};
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct DatadogLogsConfig {
+    endpoint: Option<UriSerde>,
+    api_key: String,
+    encoding: Encoding,
+    tls: Option<TlsOptions>,
+}
+
+inventory::submit! {
+    SinkDescription::new_without_default::<DatadogLogsConfig>("datadog_logs")
+}
+
+#[typetag::serde(name = "datadog_logs")]
+impl SinkConfig for DatadogLogsConfig {
+    fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let (host, port, tls) = if let Some(uri) = &self.endpoint {
+            let host = uri
+                .host()
+                .ok_or_else(|| "A host is required for endpoints".to_string())?;
+            let port = uri
+                .port_u16()
+                .ok_or_else(|| "A port is required for endpoints".to_string())?;
```

Comment:
Should these be proper `Snafu` type errors?

Reply:
I figured for such a simple sink its probably fine to just use the string here.